src/runtime/undefined.go

```diff
@@ -1,10 +1,294 @@
+func (u Undefined) Mod(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) RMod(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Eq(a any) (any, error) {
+	return reflect.TypeOf(a).Name() == reflect.TypeOf(u).Name(), nil
+}
+
+func (u Undefined) Ne(a any) (any, error) {
+	eq, err := u.Eq(a)
+	return !eq.(bool), err
+}
+
+func (u Undefined) Lt(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Le(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Gt(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Ge(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Pow(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) RPow(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Pos() (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Neg() (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Bool() (bool, error) {
+	u.logMessage()
+	return false, nil
+}
+
+func (Undefined) Repr() string {
+	return "Undefined"
+}
+
+func (u Undefined) String_() (string, error) {
+	u.logMessage()
+	return "", nil
+}
+
+func (u Undefined) Len() (int, error) {
+	return 0, nil
+}
+
+func (u Undefined) Iter() ([]any, error) {
+	return nil, nil
+}
+
+func (u Undefined) Call(...any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) GetItem(any) (any, error) {
+	return nil, u.failWithUndefinedError()
+}
+
+func (u Undefined) Int(any) (int64, error) {
+	return 0, u.failWithUndefinedError()
+}
+
+func (u Undefined) Float(any) (int64, error) {
+	return 0, u.failWithUndefinedError()
+}
+
+func (u Undefined) Complex(any) (int64, error) {
+	return 0, u.failWithUndefinedError()
+}
+
+func (u Undefined) Hash() (int64, error) {
+	h, _ := strconv.ParseInt(fmt.Sprintf("%p", &u), 0, 64)
+	return h, nil
+}
+
+func (su StrictUndefined) String_() (string, error) {
+	su.logMessage()
+	return "", su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Bool() (bool, error) {
+	su.logMessage()
+	return false, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Eq() (any, error) {
+	return nil, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Ne() (any, error) {
+	return nil, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Hash() (int64, error) {
+	return 0, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Len() (int64, error) {
+	return 0, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Iter() ([]any, error) {
+	return nil, su.failWithUndefinedError()
+}
+
+func (su StrictUndefined) Contains(any) (bool, error) {
+	return false, su.failWithUndefinedError()
+}
+
+func objectTypeRepr(a any) string {
+	return reflect.TypeOf(a).Name()
+}
+
+func (du DebugUndefined) String_() (string, error) {
```

Comment:
`String_()`?

Reply:
I had to distinguish from `func String() string` that's used, eg by fmt.Sprint